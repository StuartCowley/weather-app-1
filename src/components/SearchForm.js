import "../styles/SearchForm.css";
import React from "react";
import PropTypes from "prop-types";

const SearchForm = ({ handleCitySearch, handleCityInput }) => {
  return (
    <form className="search-form">
      <input
        className="search-form__input"
        type="text"
        onChange={handleCityInput}
      />
      <br />
      <button
        type="submit"
        className="search-form__button"
        onClick={handleCitySearch}
      >
        Search
      </button>
    </form>
  );
};

SearchForm.propTypes = {
  handleCityInput: PropTypes.func.isRequired,
  handleCitySearch: PropTypes.func.isRequired,
};

export default SearchForm;
