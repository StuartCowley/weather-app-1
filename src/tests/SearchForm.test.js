import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    handleCityInput: () => {},
    handleCitySearch: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        handleCityInput={validProps.handleCityInput}
        handleCitySearch={validProps.handleCitySearch}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByText } = render(
      <SearchForm
        handleCityInput={validProps.handleCityInput}
        handleCitySearch={validProps.handleCitySearch}
      />
    );

    expect(getByText("Search")).toHaveClass("search-form__button");
  });
});
