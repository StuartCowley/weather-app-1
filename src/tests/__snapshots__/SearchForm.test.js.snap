// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-form__input"
      id="search-form__input"
      type="text"
    />
    <br />
    <button
      class="search-form__button"
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
